sklearn/neighbors/src/nth_element.h

```diff
@@ -0,0 +1,9 @@
+#include <Python.h>
+
+int partition_node_indices(double *data,
+                           Py_intptr_t *node_indices,
+                           Py_intptr_t split_dim,
+                           Py_intptr_t split_index,
+                           Py_intptr_t n_features,
+                           Py_intptr_t n_points);
+
```

Comment:
The Scipy PR (https://github.com/scipy/scipy/pull/11653) put the comparator in the `.h` file. Do you think there would be any benefit in this?

Reply:
I switched to a template based implementation of `partition_node_indices`. It can only be place in a header file.